### ISSUE TITLE
Data sync failure

### ISSUE THREAD

**Issue by simonogunde** [dhis2/dhis2-android-capture-app#1225]
*The bug*.
There is an error occurring during synchronization of data of a program which we use DHIS capture app version 1.3.2. The problem started after changing the codes for three option sets embedded within three data elements collected within two different event tracker forms.

**Smartphone (please complete the following information):**
 - Device: Samsung galaxy A2 core
 - OS: SM-A260F
 - Browser: Google Chrome
 - Version: 8.1.0

**Comment by Balcan** [comment]
Hi @simonogunde,
Can you test with v2.0.1? We released it a couple of days ago with several fixes related to option codes. Let us kown if that is enough :)

**Comment by simonogunde** [comment]
Hi @Balcan,
Thanks for your feedback and advice. I have tested v2.0.1 but the issue has persisted. I have then noted a mistake was done during device selection whereby the mobile phones' RAM does not meet the recommended bytes (2GB minimum). Could this be a reason? Is there any other means to do data synchronization if the devices' RAM is 1GB while the recommended is 2GB?

**Comment by Balcan** [comment]
No, the only problem with RAM would be a slower performance in some parts of the app. It doesn't affect the sync process.
Is there any error log in the app settings? Or are the TEIs or Events being marked with a red sync icon? In this last scenario, clicking on it will open a list of possible errors.
If you need more help, maybe you can share some testing credentials for your server with android@dhis2.org and we will have a look.

**Comment by simonogunde** [comment]
Thanks once again. Your responses are very helpful.
Yes, there is a sync error note which shows up whenever I click to SYNC DATA NOW manually (i.e. An error has occurred during synchronization. please retry) but if I open the sync error log the page is completely blank.
Currently there is no events with a red sync icon. It previously occurred at one of the devices whereby I deleted that event. Could this perhaps be at the back-end and thus preventing other data to sync?
Thanks you.

**Comment by Balcan** [comment]
Sorry for the late response;
If none of the events have a red sync icon then it is probably a back-end error... You can try to sync the events one by one or program by program with 2.0.1 clicking on the sync icon in each item. That way you may find where the problem comes from.

**Comment by simonogunde** [comment]
Thanks Pablo. Yes I have seen that with v2.0.1 I can sync the events one by one. The data can now sync and be stored in the server but do not appear in the analytics. Am working with a colleague to sort out the matter but if you can have an idea on the cause of this we'll be so grateful.
